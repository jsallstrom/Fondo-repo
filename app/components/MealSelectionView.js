import {
  ActivityIndicator,
  Button,
  FlatList,
  Image,
  Pressable,
  SafeAreaView,
  StyleSheet,
  Text,
  View,
} from "react-native";
import React, { useEffect, useState } from "react";
import { useNavigation } from "@react-navigation/native";

const MealSelectionView = () => {
  const navigation = useNavigation();

  const [error, seterror] = useState(null);

  const [isLoading, setIsLoading] = useState(false);

  const [data, setData] = useState([]);

  const [currentLimit, setcurrentLimit] = useState(4);

  const [currentOffset, setoffset] = useState(0);

  useEffect(() => {
    fetchData(currentLimit, currentOffset);
  }, []);

  const fetchData = async (limit, offset) => {
    try {
      setIsLoading(true);
      const resp = await fetch(
        `https://playground.devskills.co/api/rest/meal-roulette-app/meals/limit/${limit}/offset/${offset}`
      );
      const data = await resp.json();
      setData(data.meal_roulette_app_meals_aggregate.nodes);
    } catch (error) {
      seterror(error);
    } finally {
      setIsLoading(false);
    }
  };

  const renderItem = ({ id, title, picture }, index) => {
    // will use id to navigate to MealDetailsScreen for specific Meal
    console.log({ title });

    return (
      <Pressable
        key={index}
        style={styles.mealItemContainer}
        // TODO: make height more dynamic
        onPress={() => navigation.navigate("MealDetailsScreen", { id: id })}
      >
        <Image style={styles.mealItemImage} source={{ uri: picture }}></Image>
        <Text style={styles.mealItemText}>{title}</Text>
      </Pressable>
    );
  };

  if (isLoading) return <ActivityIndicator size={"large"}></ActivityIndicator>;
  if (error) return <Text style={styles.errorText}>{error}</Text>;

  return (
    <SafeAreaView style={styles.container}>
      <FlatList
        data={data}
        numColumns={2}
        renderItem={({ item }, index) => {
          return renderItem(item, index);
        }}
      ></FlatList>
      <Button
        title="Refresh"
        onPress={() => {
          const newOffset = currentOffset + currentLimit;
          fetchData(currentLimit, newOffset);
          setoffset(newOffset);
        }}
      ></Button>
    </SafeAreaView>
  );
};
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },

  errorText: { color: "red" },

  mealItemContainer: {
    padding: 10,
    width: "50%",
    height: 150,
    marginBottom: 100,
  },
  mealItemImage: {
    width: "100%",
    height: "100%",
    borderRadius: 20,
  },
  mealItemText: {
    color: "white",
    textAlign: "center",
    marginTop: 30,
  },
});

export default MealSelectionView;
