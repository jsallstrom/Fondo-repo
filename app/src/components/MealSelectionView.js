import {
  Button,
  FlatList,
  Image,
  Pressable,
  StyleSheet,
  Text,
  View,
} from "react-native";
import React, { useEffect, useState } from "react";
import LoadingComponent from "./LoadingComponent";
import { API } from "../constants/API";
import ErrorMessage from "./ErrorMessage";

import Layout from "../constants/Layout";
import RefreshButton from "./RefreshButton";
import FoodItem from "./FoodItem";

const MealSelectionView = ({ navigation }) => {
  const [error, seterror] = useState(null);

  const [isLoading, setIsLoading] = useState(false);

  const [data, setData] = useState([]);

  const [currentLimit, setcurrentLimit] = useState(4);

  const [currentOffset, setoffset] = useState(0);

  useEffect(() => {
    fetchData(currentLimit, currentOffset);
  }, []);

  const fetchData = async (limit, offset) => {
    try {
      setIsLoading(true);
      const resp = await fetch(API + `limit/${limit}/offset/${offset}`);

      const data = await resp.json();

      if (data.meal_roulette_app_meals_aggregate.nodes.length === 0) {
        throw new Error("Nothing was returned from API!");
      }
      setData(data.meal_roulette_app_meals_aggregate.nodes);
      seterror(null);
    } catch (error) {
      seterror(error.message);
    } finally {
      setIsLoading(false);
    }
  };

  const renderItem = ({ id, title, picture }, index) => {
    return (
      <FoodItem
        key={index}
        title={title}
        picture={picture}
        onPress={() =>
          navigation.navigate("MealDetailsScreen", { id: id, title: title })
        }
        isSmallDevice={Layout.isSmallDevice}
      ></FoodItem>
    );
  };

  return (
    <View style={styles.container}>
      {error && <ErrorMessage message={error}></ErrorMessage>}
      {isLoading ? (
        <LoadingComponent />
      ) : (
        <FlatList
          data={data}
          numColumns={Layout.isSmallDevice ? 1 : 2}
          renderItem={({ item }, index) => {
            return renderItem(item, index);
          }}
          testID={"food-flatlist"}
        ></FlatList>
      )}

      <RefreshButton
        onPress={() => {
          const newOffset = currentOffset + currentLimit;
          fetchData(currentLimit, newOffset);
          setoffset(newOffset);
        }}
        title={"Refresh"}
      ></RefreshButton>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

export default MealSelectionView;
