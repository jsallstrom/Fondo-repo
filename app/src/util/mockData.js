export const mockData1 = {
  nodes: [
    {
      id: 1,
      title: "Chorizo & mozzarella gnocchi bake",
      picture:
        "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Chorizo%20%26%20mozzarella%20gnocchi%20bake.webp?alt=media&token=d8511cac-4029-4c44-a5bc-35b8b0664cfe",
    },
    {
      id: 2,
      title: "Easy butter chicken",
      picture:
        "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20butter%20chicken.webp?alt=media&token=baaf3b5b-0528-4bcf-b5c0-d429c95313ed",
    },
    {
      id: 4,
      title: "Easy classic lasagne",
      picture:
        "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20classic%20lasagne.webp?alt=media&token=c13ea24f-cff5-44c3-90a9-f5f3e0b47e90",
    },
    {
      id: 5,
      title: "Thai fried prawn & pineapple rice",
      picture:
        "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Thai%20fried%20prawn%20%26%20pineapple%20rice.webp?alt=media&token=268c55fc-e977-496c-be36-b99d9e6eba04",
    },
  ],
};

export const mockData2 = {
  nodes: [
    {
      id: 6,
      title: "One-pan spaghetti with nduja, fennel & olives",
      picture:
        "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/One-pan%20spaghetti.webp?alt=media&token=8a36711c-e986-4742-a485-08ea71447109",
    },
    {
      id: 7,
      title: "Easy pancakes",
      picture:
        "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20pancakes.webp?alt=media&token=3b8df6fd-63e5-4661-805c-4868179487b4",
    },
    {
      id: 8,
      title: "Easy chicken fajitas",
      picture:
        "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/%20Easy%20chicken%20fajitas.webp?alt=media&token=4bc04b16-5c96-42f0-aa6d-4f52f1d11a91",
    },
    {
      id: 9,
      title: "Easy carrot cake",
      picture:
        "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20carrot%20cake.webp?alt=media&token=d55b4b7d-0639-4b89-b4ea-b4e96dbf4f96",
    },
  ],
};
