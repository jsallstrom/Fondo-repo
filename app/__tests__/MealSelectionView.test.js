import React from "react";

import { render, fireEvent, act } from "@testing-library/react-native";
import MealSelectionView from "../src/components/MealSelectionView";
import { mockData1, mockData2 } from "../src/util/mockData";

test("render component and expect to match snapshot", async () => {
  fetch.mockResponseOnce(
    // If the fetch is immediately when you load component, you have to set the mocked fetch before you load component
    JSON.stringify({ meal_roulette_app_meals_aggregate: mockData1 })
  ); // when fetch is called this is what will be returned
  const navigate = jest.fn();

  const component = render(<MealSelectionView navigation={{ navigate }} />);

  await act(() => new Promise((resolve) => setImmediate(resolve))); // wait for the fetch to be resolved

  expect(component).toMatchSnapshot();
});

test("before the component has loaded the isLoadingcomponent should be shown", async () => {
  fetch.mockResponses([
    JSON.stringify({ meal_roulette_app_meals_aggregate: mockData1 }),
  ]);

  const navigate = jest.fn();

  const component = render(<MealSelectionView navigation={{ navigate }} />);

  const { queryByTestId } = component;

  const loader = queryByTestId("loader");

  expect(loader).not.toBeNull();
  await act(() => new Promise((resolve) => setImmediate(resolve)));
});

test("pressing the refresh button, refreshes the page", async () => {
  fetch.mockResponses(
    [JSON.stringify({ meal_roulette_app_meals_aggregate: mockData1 })],
    [JSON.stringify({ meal_roulette_app_meals_aggregate: mockData2 })]
  );

  const navigate = jest.fn();

  const component = render(<MealSelectionView navigation={{ navigate }} />);

  const { getByText } = component;
  await act(() => new Promise((resolve) => setImmediate(resolve)));

  const refreshButton = getByText("Refresh");

  fireEvent.press(refreshButton);

  await act(() => new Promise((resolve) => setImmediate(resolve)));

  expect(component).toMatchSnapshot();
});

test("pressing the refresh button until no more meals arrive gives error message", async () => {
  fetch.mockResponses([
    JSON.stringify({ meal_roulette_app_meals_aggregate: mockData1 }),
  ]);

  const navigate = jest.fn();

  const component = render(<MealSelectionView navigation={{ navigate }} />);

  const { getByText, queryByTestId } = component;
  await act(() => new Promise((resolve) => setImmediate(resolve)));

  const refreshButton = getByText("Refresh");

  fireEvent.press(refreshButton);

  await act(() => new Promise((resolve) => setImmediate(resolve)));

  const errorText = queryByTestId("errorText");

  expect(errorText).not.toBeNull();
});

test("navigate should be called when pressing on each of the food items", async () => {
  fetch.mockResponses([
    JSON.stringify({ meal_roulette_app_meals_aggregate: mockData1 }),
  ]);

  const navigate = jest.fn();

  const component = render(<MealSelectionView navigation={{ navigate }} />);

  const { debug, getAllByTestId } = component;
  await act(() => new Promise((resolve) => setImmediate(resolve)));

  const foodFlatlist = getAllByTestId("food-item");

  for (let i = 0; i < foodFlatlist.length; i++) {
    const foodItemPressable = foodFlatlist[i];
    fireEvent.press(foodItemPressable);
  }

  expect(navigate.mock.calls.length).toBe(4); // expect each food item to be pressed
});
