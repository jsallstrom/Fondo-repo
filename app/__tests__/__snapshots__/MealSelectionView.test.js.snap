// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pressing the refresh button, refreshes the page 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 6,
          "picture": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/One-pan%20spaghetti.webp?alt=media&token=8a36711c-e986-4742-a485-08ea71447109",
          "title": "One-pan spaghetti with nduja, fennel & olives",
        },
        Object {
          "id": 7,
          "picture": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20pancakes.webp?alt=media&token=3b8df6fd-63e5-4661-805c-4868179487b4",
          "title": "Easy pancakes",
        },
        Object {
          "id": 8,
          "picture": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/%20Easy%20chicken%20fajitas.webp?alt=media&token=4bc04b16-5c96-42f0-aa6d-4f52f1d11a91",
          "title": "Easy chicken fajitas",
        },
        Object {
          "id": 9,
          "picture": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20carrot%20cake.webp?alt=media&token=d55b4b7d-0639-4b89-b4ea-b4e96dbf4f96",
          "title": "Easy carrot cake",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="food-flatlist"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 333.5,
                "marginBottom": 100,
                "padding": 10,
                "width": "50%",
              }
            }
            testID="food-item"
          >
            <Image
              source={
                Object {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/One-pan%20spaghetti.webp?alt=media&token=8a36711c-e986-4742-a485-08ea71447109",
                }
              }
              style={
                Object {
                  "borderRadius": 20,
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
            <Text
              style={
                Object {
                  "marginTop": 30,
                  "textAlign": "center",
                }
              }
            >
              One-pan spaghetti with nduja, fennel & olives
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 333.5,
                "marginBottom": 100,
                "padding": 10,
                "width": "50%",
              }
            }
            testID="food-item"
          >
            <Image
              source={
                Object {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20pancakes.webp?alt=media&token=3b8df6fd-63e5-4661-805c-4868179487b4",
                }
              }
              style={
                Object {
                  "borderRadius": 20,
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
            <Text
              style={
                Object {
                  "marginTop": 30,
                  "textAlign": "center",
                }
              }
            >
              Easy pancakes
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 333.5,
                "marginBottom": 100,
                "padding": 10,
                "width": "50%",
              }
            }
            testID="food-item"
          >
            <Image
              source={
                Object {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/%20Easy%20chicken%20fajitas.webp?alt=media&token=4bc04b16-5c96-42f0-aa6d-4f52f1d11a91",
                }
              }
              style={
                Object {
                  "borderRadius": 20,
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
            <Text
              style={
                Object {
                  "marginTop": 30,
                  "textAlign": "center",
                }
              }
            >
              Easy chicken fajitas
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 333.5,
                "marginBottom": 100,
                "padding": 10,
                "width": "50%",
              }
            }
            testID="food-item"
          >
            <Image
              source={
                Object {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20carrot%20cake.webp?alt=media&token=d55b4b7d-0639-4b89-b4ea-b4e96dbf4f96",
                }
              }
              style={
                Object {
                  "borderRadius": 20,
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
            <Text
              style={
                Object {
                  "marginTop": 30,
                  "textAlign": "center",
                }
              }
            >
              Easy carrot cake
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#2196F3",
        "borderRadius": 100,
        "elevation": 5,
        "height": 148.22222222222223,
        "justifyContent": "center",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
        "width": 148.22222222222223,
      }
    }
    title="Refresh"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
      Refresh
    </Text>
  </View>
</View>
`;

exports[`render component and expect to match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 1,
          "picture": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Chorizo%20%26%20mozzarella%20gnocchi%20bake.webp?alt=media&token=d8511cac-4029-4c44-a5bc-35b8b0664cfe",
          "title": "Chorizo & mozzarella gnocchi bake",
        },
        Object {
          "id": 2,
          "picture": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20butter%20chicken.webp?alt=media&token=baaf3b5b-0528-4bcf-b5c0-d429c95313ed",
          "title": "Easy butter chicken",
        },
        Object {
          "id": 4,
          "picture": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20classic%20lasagne.webp?alt=media&token=c13ea24f-cff5-44c3-90a9-f5f3e0b47e90",
          "title": "Easy classic lasagne",
        },
        Object {
          "id": 5,
          "picture": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Thai%20fried%20prawn%20%26%20pineapple%20rice.webp?alt=media&token=268c55fc-e977-496c-be36-b99d9e6eba04",
          "title": "Thai fried prawn & pineapple rice",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="food-flatlist"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 333.5,
                "marginBottom": 100,
                "padding": 10,
                "width": "50%",
              }
            }
            testID="food-item"
          >
            <Image
              source={
                Object {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Chorizo%20%26%20mozzarella%20gnocchi%20bake.webp?alt=media&token=d8511cac-4029-4c44-a5bc-35b8b0664cfe",
                }
              }
              style={
                Object {
                  "borderRadius": 20,
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
            <Text
              style={
                Object {
                  "marginTop": 30,
                  "textAlign": "center",
                }
              }
            >
              Chorizo & mozzarella gnocchi bake
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 333.5,
                "marginBottom": 100,
                "padding": 10,
                "width": "50%",
              }
            }
            testID="food-item"
          >
            <Image
              source={
                Object {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20butter%20chicken.webp?alt=media&token=baaf3b5b-0528-4bcf-b5c0-d429c95313ed",
                }
              }
              style={
                Object {
                  "borderRadius": 20,
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
            <Text
              style={
                Object {
                  "marginTop": 30,
                  "textAlign": "center",
                }
              }
            >
              Easy butter chicken
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 333.5,
                "marginBottom": 100,
                "padding": 10,
                "width": "50%",
              }
            }
            testID="food-item"
          >
            <Image
              source={
                Object {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Easy%20classic%20lasagne.webp?alt=media&token=c13ea24f-cff5-44c3-90a9-f5f3e0b47e90",
                }
              }
              style={
                Object {
                  "borderRadius": 20,
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
            <Text
              style={
                Object {
                  "marginTop": 30,
                  "textAlign": "center",
                }
              }
            >
              Easy classic lasagne
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "height": 333.5,
                "marginBottom": 100,
                "padding": 10,
                "width": "50%",
              }
            }
            testID="food-item"
          >
            <Image
              source={
                Object {
                  "uri": "https://firebasestorage.googleapis.com/v0/b/devskills-prod.appspot.com/o/Thai%20fried%20prawn%20%26%20pineapple%20rice.webp?alt=media&token=268c55fc-e977-496c-be36-b99d9e6eba04",
                }
              }
              style={
                Object {
                  "borderRadius": 20,
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
            <Text
              style={
                Object {
                  "marginTop": 30,
                  "textAlign": "center",
                }
              }
            >
              Thai fried prawn & pineapple rice
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#2196F3",
        "borderRadius": 100,
        "elevation": 5,
        "height": 148.22222222222223,
        "justifyContent": "center",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
        "width": 148.22222222222223,
      }
    }
    title="Refresh"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
      Refresh
    </Text>
  </View>
</View>
`;
